### DISCUSSION TITLE
AppleScript: How to run this cross-seed?

### DISCUSSION THREAD

**Discussion by SawkeeReemo** [cross-seed/cross-seed#258] [Q&A] [answered]
I don't know if this is the best place to ask this, but I figured I would ask here after searching for a solution for a couple hours:

When I type `/usr/local/bin/cross-seed search` into the Terminal (macOS) cross-seed runs as expected. 

But when I add `do shell script "/usr/local/bin/cross-seed search"` to an AppleScript, I get: 
`error "env: node: No such file or directory" number 127`

Any idea how to make this work?

**Comment by mmgoodnow** [comment, marked as answer]
It looks like it can't find the node executable, which means it's either not in your PATH or you don't have a PATH set inside the AppleScript environment. In your terminal, `which node` should give you the location of the executable which you can then prepend the cross-seed location with. So it'd be `/path/to/node /path/to/cross/seed [options]`

**Reply by SawkeeReemo** [reply]
Ya know, it’s funny, I found that solution elsewhere as well, but they said to do `type node` to get the path to node. But what it gave me was wrong. I found it manually and added it to my script, and it works. So I’m hoping `which node` will give me the correct path so I can make it a variable. 

Your cross-seed app, and my Synology’s strange lack of being able to upgrade node.js has really got me into scripting. So thanks! but also… damn you! 🤣
